client/internal/dns/host_darwin.go

```diff
@@ -165,13 +165,13 @@ func (s *systemConfigurator) removeKeyFromSystemConfig(key string) error {
 }
 
 func (s *systemConfigurator) addLocalDNS() error {
-	if s.systemDNSSettings.ServerIP == "" || len(s.systemDNSSettings.Domains) == 0 {
```

Comment:
`ServerIP.IsValid()` or `!ServerIP.IsValid()` is correct here? previous state was `ServerIP == ""`